src/main/kotlin/com/team4099/robot2023/subsystems/superstructure/Superstructure.kt

```diff
@@ -0,0 +1,990 @@
+                when (usingGamePiece) {
+                  GamePiece.CUBE ->
+                    Elevator.TunableElevatorHeights.hybridHeight.get() +
+                      Elevator.TunableElevatorHeights.cubeDropPosition.get()
+                  GamePiece.CONE ->
+                    Elevator.TunableElevatorHeights.hybridHeight.get() +
+                      Elevator.TunableElevatorHeights.coneDropPosition.get()
+                  else -> 0.0.inches
+                }
+              }
+              NodeTier.MID -> {
+                when (usingGamePiece) {
+                  GamePiece.CUBE ->
+                    Elevator.TunableElevatorHeights.midCubeHeight.get() +
+                      Elevator.TunableElevatorHeights.cubeDropPosition.get()
+                  GamePiece.CONE ->
+                    Elevator.TunableElevatorHeights.midConeHeight.get() +
+                      Elevator.TunableElevatorHeights.coneDropPosition.get()
+                  else -> 0.0.inches
+                }
+              }
+              NodeTier.HIGH -> {
+                when (usingGamePiece) {
+                  GamePiece.CUBE ->
+                    Elevator.TunableElevatorHeights.highCubeHeight.get() +
+                      Elevator.TunableElevatorHeights.cubeDropPosition.get()
+                  GamePiece.CONE ->
+                    Elevator.TunableElevatorHeights.highConeHeight.get() +
+                      Elevator.TunableElevatorHeights.coneDropPosition.get()
+                  else -> 0.0.inches
+                }
+              }
+              else -> 0.0.inches
+            }
+          val extension =
+            when (nodeTier) {
+              NodeTier.HYBRID -> Manipulator.TunableManipulatorStates.lowScoreExtension.get()
+              NodeTier.MID -> Manipulator.TunableManipulatorStates.midScoreExtension.get()
+              NodeTier.HIGH -> Manipulator.TunableManipulatorStates.highScoreExtension.get()
+              else -> 0.0.inches
+            }
+          manipulator.currentRequest =
+            Request.ManipulatorRequest.TargetingPosition(
+              extension, ManipulatorConstants.IDLE_VOLTAGE
+            )
+          elevator.currentRequest = Request.ElevatorRequest.TargetingPosition(scoreHeight)
+        }
+
+        // Transition
+        if (currentRequest is SuperstructureRequest.Score &&
+          groundIntake.isAtTargetedPosition &&
+          manipulator.isAtTargetedPosition &&
+          elevator.isAtTargetedPosition
+        ) {
+          nextState =
+            when (usingGamePiece) {
+              GamePiece.CUBE -> SuperstructureStates.SCORE_CUBE
+              GamePiece.CONE -> SuperstructureStates.SCORE_CONE
+              else -> SuperstructureStates.IDLE
+            }
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.IDLE
+        }
+      }
+      SuperstructureStates.SCORE_CUBE -> {
+        // Outputs
+        manipulator.currentRequest =
```

Comment:
similarly to the prep state: you need to retract the manipulator before moving the elevator down, otherwise it will clip the lower nodes as the elevator retracts